documentation/EN/chapter_11.md

```diff
@@ -0,0 +1,956 @@
+# 11. AODVv2 protocol initialization operations.
+We will explain the processes involved in the initialization and start-up of the _AODVv2_ protocol about the embedded hardware based on the _cc1312_ and the _RIOT-OS_ operating system, which make up the **Turpial** platform.
+
+## 11.1 Initialization process.
+
+<p>
+<img src="imple_pic/aodvv2_init.svg" alt="drawing" height="500" width="300" align="left"/>
+</p>
+
+The initialization process of the _AODV_ protocol on a network node requires following a series of steps to configure the software and hardware necessary for the correct operation of the routing protocol.
+
+In the image we can see every function that must be executed before we can connect to the **Locha Mesh** network.
+
+The processes executed in the initialization stage configure the tables managed within the protocol, initialize the network interface and obtain the necessary IP address to operate on the network, among other resources.
+
+All algorithms included here are developed under the _RIOT-OS_ operating system by the **Locha** team.
+
+
+<br />&nbsp;<br />
+<br />&nbsp;<br />
+<br />&nbsp;<br />
+<br />&nbsp;<br />
+<br />&nbsp;<br />
+
+
+
+## 11.2 aodvv2_seqnum_init
+
+As specified in chapter 8.9, the sequence number allows us to know if the route messages are obsolete or not, and keep fresh route information at all times.
+
+The _AODVv2_seqnum_init_ function is part of some functions located inside a file called _aodvv2_seqnum.c_, with the following functions:
+
+- aodvv2_seqnum_init(void).
+- aodvv2_seqnum_inc(void).
+- aodvv2_seqnum_get().
+
+Knowing the role that the sequence number plays in the _AODV_ protocol, we see the functions do no more than what their name says:
+
+ - Starting a counter in **1** (init), as detailed in the protocol description.
+
+ - Read the counter and get its value. (get)
+
+ - Increase the counter by one (inc), as detailed in the protocol, for every _RREQ_ or _RREP_ that this node generates.
+
+
+
+### 11.2.1 Routine to handle the sequence number.
+
+```cpp
+#include "net/aodvv2/seqnum.h"
+
+#include <stdatomic.h>
+
+static atomic_uint_fast32_t seqnum;
+
+void aodvv2_seqnum_init(void)
+{
+    /* Initialize to 1 */
+    atomic_init(&seqnum, 1);
+}
+
+void aodvv2_seqnum_inc(void)
+{
+    if (atomic_fetch_add(&seqnum, 1) >= 65535) {
+        atomic_store(&seqnum, 1);
+    }
+}
+
+aodvv2_seqnum_t aodvv2_seqnum_get(void)
+{
+    return atomic_load(&seqnum);
+}
+
+```
+
+The library is very simple, but efficiently it solves possible data problems that can be detected when different execution threads in the application try to read and write the object that represents the sequence  number.
+
+## 11.3 aodvv2_routing_table_init
+
+Todos los routers AODV deben mantener un conjunto de rutas locales, conteniendo información obtenida de los mensajes de ruta (RREQ y RREP).
```

Comment:
This is in Spanish, it must be in English.

Reply:
updated, thanks 👍